client/src/components/CompanyRegister.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from "react";
+import {
+  Form,
+  Input,
+  Dropdown,
+  Button,
+  TextArea,
+  Message,
+  Grid,
+  Header
+} from "semantic-ui-react";
+
+import { companyRegister } from "../api/companies";
+import { industryList } from "../utils/constants";
+
+class CompanyRegister extends Component {
+  state = {
+    companyProfile: {
+      name: "",
+      description: "",
+      industry: [],
+      email: "",
+      password: "",
+      user_id: ""
+    },
+    error: null,
+    success: null
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    companyRegister(this.state.companyProfile)
+      .then(data => {
+        console.log(data);
+        if (data.success) {
+          this.setState({
+            companyProfile: {
+              name: "",
+              description: "",
+              industry: [],
+              email: "",
+              password: ""
+            },
+            success: true
+          });
+        } else {
+          this.setState({});
+        }
+        this.resetForm(data.success);
+      })
+      .catch(err => {
+        this.setState({ success: false, error: err });
```

Comment:
it's better to return a string (currently you returning an object), for example, 
```this.setState({ err.message ||  'error' });```